### PR TITLE
chore: Changed token_count to only use tokenCountCallback

### LINKED ISSUE #271: bedrock: Only assign token_count from result of tokenCountCallback
### Description
We realized token counts will not be accurate for many use cases.  Token counts are typically for all prompts and responses and not on the individual message. In order to simplify logic and not expose edge cases in calculation we are going to rely completely on token counts getting calculated downstream in the pipeline. In cases where content will not be provided, customers can register a callback and we must call it to get the token count

### Acceptance Criteria
If the customer provides the `tokenCountCallback` then the agent **MUST** call the callback to get the input token counts, otherwise the agent **MUST** omit the attribute from the event and the token counts will be attached server side.

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolves #271.

### CODE REVIEW

File: lib/llm/bedrock-response.js [hunk 1]
```diff
@@ -122,25 +122,6 @@ class BedrockResponse {
     return this.#id
   }
 
-  /**
-   * The number of tokens present in the prompt as determined by the remote
-   * API.
-   *
-   * @returns {number|undefined}
-   */
-  get inputTokenCount() {
```
Comment (bizob2828) [inline lib/llm/bedrock-response.js:131]: we can't remove this yet as it's still assigning the token counts on the summary. The cleanup of that is in #265

File: tests/versioned/common.js [hunk 2]
```diff
@@ -146,15 +139,6 @@ function assertChatCompletionSummary({
     'error': error
   }
 
-  if (tokenUsage) {
```
Comment (bizob2828) [inline tests/versioned/common.js:149]: we still need this. it's being removed in #265

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: tests/versioned/v3/bedrock-chat-completions.tap.js [hunk 3]
```diff
@@ -189,7 +189,7 @@ tap.afterEach(async (t) => {
         chatMsgs
       })
 
-      t.llmSummary({ tx, modelId, chatSummary, tokenUsage: true, numMsgs: events.length - 1 })
+      t.llmSummary({ tx, modelId, chatSummary, numMsgs: events.length - 1 })
```
Comment (bizob2828) [inline tests/versioned/v3/bedrock-chat-completions.tap.js:192]: this should be restored

File: tests/unit/llm/bedrock-response.tap.js [hunk 4]
```diff
@@ -98,8 +98,6 @@ tap.test('non-conforming response is handled gracefully', async (t) => {
   t.equal(res.finishReason, undefined)
   t.same(res.headers, undefined)
   t.equal(res.id, undefined)
-  t.equal(res.inputTokenCount, undefined)
```
Comment (bizob2828) [inline tests/unit/llm/bedrock-response.tap.js:101]: this should be restored

File: tests/unit/llm/stream-handler.tap.js [hunk 5]
```diff
@@ -117,8 +117,6 @@ tap.test('handles claude streams', async (t) => {
   t.equal(br.finishReason, 'done')
   t.equal(br.requestId, 'aws-req-1')
   t.equal(br.statusCode, 200)
-  t.equal(br.inputTokenCount, 5)
```
Comment (bizob2828) [inline tests/unit/llm/stream-handler.tap.js:120]: this should be restored

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: tests/unit/llm/bedrock-response.tap.js (hunk 4 above)
Comment (bizob2828) [reply tests/unit/llm/bedrock-response.tap.js:101]: i meant the entire file, sorry for the confusion

File: tests/unit/llm/stream-handler.tap.js (hunk 5 above)
Comment (bizob2828) [reply tests/unit/llm/stream-handler.tap.js:120]: i meant the entire file, sorry for the confusion

File: tests/unit/llm/bedrock-response.tap.js (hunk 4 above)
Comment (jsumners-nr) [reply tests/unit/llm/bedrock-response.tap.js:101]: Why? What are we trying to test by doing so?

Comment (bizob2828) [reply tests/unit/llm/bedrock-response.tap.js:101]: i understand this code is going away in #265. so it's probably fine to leave as is but the test coverage will be missing until we rip it all out.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**